internal/pkg/describe/workload.go

```diff
@@ -29,34 +30,57 @@ type workloadStackDescriber struct {
 	stackResources []*stack.Resource
 }
 
-type workloadConfig struct {
-	app         string
-	name        string
-	configStore ConfigStoreSvc
+// NewWorkloadConfig contains fields that initiates workload describer struct.
+type NewWorkloadConfig struct {
+	App         string
+	Env         string
+	Name        string
+	ConfigStore ConfigStoreSvc
 }
 
-// newWorkloadStackDescriber instantiates the core elements of a new workload.
-func newWorkloadStackDescriber(opt workloadConfig, env string) (*workloadStackDescriber, error) {
-	environment, err := opt.configStore.GetEnvironment(opt.app, env)
+// NewWorkloadStackDescriber instantiates the core elements of a new workload.
+func NewWorkloadStackDescriber(opt NewWorkloadConfig) (*WorkloadStackDescriber, error) {
+	environment, err := opt.ConfigStore.GetEnvironment(opt.App, opt.Env)
 	if err != nil {
-		return nil, fmt.Errorf("get environment %s: %w", env, err)
+		return nil, fmt.Errorf("get environment %s: %w", opt.Env, err)
 	}
 	sess, err := sessions.ImmutableProvider().FromRole(environment.ManagerRoleARN, environment.Region)
 	if err != nil {
 		return nil, err
 	}
-	return &workloadStackDescriber{
-		app:  opt.app,
-		name: opt.name,
-		env:  env,
+	return &WorkloadStackDescriber{
+		app:  opt.App,
+		name: opt.Name,
+		env:  opt.Env,
 
-		cfn:  stack.NewStackDescriber(cfnstack.NameForWorkload(opt.app, env, opt.name), sess),
+		cfn:  stack.NewStackDescriber(cfnstack.NameForWorkload(opt.App, opt.Env, opt.Name), sess),
 		sess: sess,
 	}, nil
 }
 
+// Version returns the CloudFormation template version associated with
+// the workload by reading the Metadata.Version field from the template.
+//
+// If the Version field does not exist, then it's a legacy template and it returns an version.LegacyWorkloadTemplate and nil error.
+func (d *WorkloadStackDescriber) Version() (string, error) {
+	raw, err := d.cfn.StackMetadata()
```

Comment:
what happens if the service hasn't been deployed yet? doesn't this return an error, when we actually want them to be able to deploy the template?

Reply:
Good point we need to address this issue

Reply:
Done